### PR TITLE
add list-listens command

### PR DESCRIPTION
**Pull request by Lederstrumpf**

### CODE REVIEW

File: src/rpc/request.rs [hunk 1]
```diff
@@ -400,6 +400,9 @@ pub enum Request {
     // #[api(type = 105)]
     // #[display("list_offer_ids()")]
     // ListOfferIds,
+    #[api(type = 105)]
+    #[display("list_offers()")]
```
Comment (TheCharlatan) [inline src/rpc/request.rs:404]: `list_listens` so we get the correct `debug` level log for the request.

**Comment by zkao** [comment]
I can just block creating two offers in the same port, if that is the whole purpose of this PR

**Comment by zkao** [comment]
with one line change, most likely

**Comment by zkao** [comment]
but i see this could be useful for other things

**Review by zkao — APPROVED** [review]

File: src/rpc/request.rs (hunk 1 above)
Comment (Lederstrumpf) [reply src/rpc/request.rs]: thanks!

File: src/rpc/request.rs [hunk 2]
```diff
@@ -531,7 +531,7 @@ pub enum Request {
     #[api(type = 1107)]
     #[display("offer_list({0})", alt = "{0:#}")]
     #[from]
-    ListenList(List<RemoteSocketAddr>),
+    ListenList(List<String>),
```
[the remark this answers is not in the document]
Comment (TheCharlatan) [reply src/rpc/request.rs:534]: Why did you change this back to `<String>` @Lederstrumpf ?

[the remark this answers is not in the document]
Comment (zkao) [reply src/rpc/request.rs:534]: he's printing urls,  which are the correct way to reconstruct this addresses if i remember correctly

h4sh3d used something else for the offer

[the remark this answers is not in the document]
Comment (Lederstrumpf) [reply src/rpc/request.rs:534]: Yup, it's urls - I reconstruct them over [here](https://github.com/farcaster-project/offer-refiller/blob/main/src/main.rs#L171-L176)

**Pull request merged**